tools/congestion-model/src/strategy/smooth_traffic_light.rs

```diff
@@ -0,0 +1,352 @@
+        let receiver = ctx.tx_receiver(tx);
+
+        let CongestedShardsInfo { congestion_level, .. } = self.get_info(ctx, &receiver);
+        congestion_level > self.reject_tx_congestion_limit
+    }
+
+    // Step 4: Execute receipts in the order of local, delayed, incoming
+    // Step 5: Remaining local or incoming receipts are added to the end of the
+    // delayed receipts queue
+    //
+    // In the model there is no distinction between local, delayed and incoming.
+    // All of those are stored in the incoming queue so we just process that.
+    //
+    // Always process as many receipts as allowed by the GAS_LIMIT.
+    //
+    // The outgoing receipts are processed as in `process_outgoing_receipts`.
+    fn process_incoming_receipts(&mut self, ctx: &mut ChunkExecutionContext<'_>) {
+        while ctx.gas_burnt() < GAS_LIMIT {
+            let Some(receipt) = ctx.incoming_receipts().pop_front() else {
+                break;
+            };
+
+            let outgoing = ctx.execute_receipt(receipt);
+            for receipt in outgoing {
+                self.forward_or_buffer(ctx, receipt);
+            }
+        }
+    }
+
+    // Step 6: Compute own congestion information for the next block
+    fn update_block_info(&mut self, ctx: &mut ChunkExecutionContext<'_>) {
+        let incoming_gas_congestion = self.incoming_gas_congestion(ctx);
+        let outgoing_gas_congestion = self.outgoing_gas_congestion(ctx);
+        let memory_congestion = self.memory_congestion(ctx);
+
+        // Initial traffic light:
+        // Usually, signal other shards to slow down based on our incoming gas congestion only.
+        // All other limits (outgoing gas, memory) are ignored until they hit a red light.
+        // The goal is to never hit red, unless we have to prevent the unbounded queues vs deadlocks tradeoff.
+        // If we hit red, it slows down incoming traffic massively, which should bring us out of red soon.
+        // let red =
+        //     incoming_gas_congestion.max(outgoing_gas_congestion).max(memory_congestion) >= 1.0;
+        // let info = if red {
+        //     CongestedShardsInfo {
+        //         congestion_level: 1.0,
+        //         allowed_shard: Some(self.round_robin_shard(ctx.block_height() as usize)),
+        //     }
+        // } else {
+        //     CongestedShardsInfo { congestion_level: incoming_gas_congestion, allowed_shard: None }
+        // };
+
+        // Simplified and smoothed. Requires a larger max memory limit but will
+        // smoothly reduce based on size, which can even lead to smaller peak
+        // buffer size.
+        let max_congestion =
+            incoming_gas_congestion.max(outgoing_gas_congestion).max(memory_congestion);
+        let info = if max_congestion >= 1.0 {
+            CongestedShardsInfo {
+                congestion_level: 1.0,
+                allowed_shard: Some(self.round_robin_shard(ctx.block_height() as usize)),
```

Comment:
Totally fine in the model but I wonder what should we use for the seed in production. 

We could do something determinictic (vrf) to get rid of it from the congestion info or just set it to a random value given in this case the chunk producer can decide on what it is. The latter gives some flexibility for both abuse and optimizations.

Reply:
What is the problem with using the block height in production?